src/axom/inlet/Inlet.hpp

```diff
@@ -194,80 +194,69 @@ class Inlet : public SchemaCreator
-   * value. 
-   *
-   * \param [in] name Name of the Field value to be gotten
-   * \param [out] value Value to be filled
-   *
-   * \return True if the value was found in the Datastore
-   *****************************************************************************
-   */
-  bool get(const std::string& name, double& value)
+ *******************************************************************************
+ * \brief Gets a value of arbitrary type out of the datastore
+ * 
+ * Retrieves a value of user-defined type.
+ * 
+ * \param [in] name The name of the subtable representing the root of the object
+ * \return The retrieved value
+ * \tparam The type to retrieve
+ * \pre Requires a specialization of from_inlet<T>(axom::inlet::Table&)
+ * \note This function does not indicate failure
+ *******************************************************************************
+ */
+  template <typename T>
+  T get(const std::string& name)
   {
-    return m_globalTable->get(name, value);
+    return m_globalTable->get<T>(name);
   }
 
   /*!
    *****************************************************************************
-   * \brief Gets a Integer value out of the Datastore.
-   *
-   * Retrieves the Field value out of the DataStore.  This Field may not have
-   * been actually present in the input file and will be indicted by the return
-   * value. 
+   * \brief Return whether a subobject with the given name is present in 
+   * the datastore.
    *
-   * \param [in] name Name of the Field value to be gotten
-   * \param [out] value Value to be filled
-   *
-   * \return True if the value was found in the Datastore
+   * \see Table::contains
    *****************************************************************************
    */
-  bool get(const std::string& name, int& value)
+  bool contains(const std::string& name)
   {
-    return m_globalTable->get(name, value);
+    return m_globalTable->contains(name);
   }
 
   /*!
-   *****************************************************************************
-   * \brief Gets a String value out of the Datastore.
-   *
-   * Retrieves the Field value out of the DataStore.  This Field may not have
-   * been actually present in the input file and will be indicted by the return
-   * value. 
-   *
-   * \param [in] name Name of the Field value to be gotten
-   * \param [out] value Value to be filled
-   *
-   * \return True if the value was found in the Datastore
-   *****************************************************************************
+   *******************************************************************************
+   * \brief Obtains a proxy view into the datastore.
+   * 
+   * \see Table::operator[]
+   *******************************************************************************
    */
-  bool get(const std::string& name, std::string& value)
-  {
-    return m_globalTable->get(name, value);
-  }
+  Proxy operator[](const std::string& name) { return (*m_globalTable)[name]; }
```

Comment:
Perhaps there is a better user-facing name for the `Proxy` type, e.g. `inlet::FieldView` or `inlet::FieldProxy` (or something better)?

Reply:
I agree that it may not be immediately clear what a `Proxy` is for - though the name is used by the `json` and `sol` libraries for roughly the same thing.  I'd be hesitant to put `Field` in the name since it can also wrap a `Table`. Maybe `SubObjectView`?